sql/core/src/main/scala/org/apache/spark/sql/execution/command/AnalyzeTableCommand.scala

```diff
@@ -88,14 +90,30 @@ case class AnalyzeTableCommand(tableName: String) extends RunnableCommand {
             }
           }.getOrElse(0L)
 
-        // Update the Hive metastore if the total size of the table is different than the size
+        val needUpdate = new mutable.HashMap[String, String]()
+        if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+          needUpdate += (AnalyzeTableCommand.TOTAL_SIZE_FIELD -> newTotalSize.toString)
+        }
+        if (!noscan) {
+          val oldRowCount = tableParameters.get(AnalyzeTableCommand.ROW_COUNT).map(_.toLong)
+            .getOrElse(-1L)
+          val newRowCount = sparkSession.table(tableName).count()
+
+          if (newRowCount >= 0 && newRowCount != oldRowCount) {
+            needUpdate += (AnalyzeTableCommand.ROW_COUNT -> newRowCount.toString)
+          }
+        }
+        // Update the Hive metastore if the above parameters of the table is different than those
         // recorded in the Hive metastore.
         // This logic is based on org.apache.hadoop.hive.ql.exec.StatsTask.aggregateStats().
-        if (newTotalSize > 0 && newTotalSize != oldTotalSize) {
+        if (needUpdate.nonEmpty) {
+          // need to set this parameter so that we can store other parameters like "numRows" into
+          // Hive metastore
+          needUpdate.put(AnalyzeTableCommand.STATS_GENERATED_VIA_STATS_TASK,
+            AnalyzeTableCommand.TRUE)
           sessionState.catalog.alterTable(
             catalogTable.copy(
-              properties = relation.catalogTable.properties +
-                (AnalyzeTableCommand.TOTAL_SIZE_FIELD -> newTotalSize.toString)))
+              properties = relation.catalogTable.properties ++ needUpdate))
```

Comment:
Just tried it in `Hive`. 

When the `ALTER TABLE` command does not set `STATS_GENERATED_VIA_STATS_TASK`, the `COLUMN_STATS_ACCURATE` is set to `false`. For example, 
```SQL
alter table t1 set TBLPROPERTIES ('numRows'='2');
```

The table properties will be like 
```
Table Parameters:	 	 
	COLUMN_STATS_ACCURATE	false               
	last_modified_by    	root                
	last_modified_time  	1471635353          
	numFiles            	1                   
	numRows             	-1                  
	rawDataSize         	-1                  
	totalSize           	2                   
	transient_lastDdlTime	1471635353     
```

If we set `STATS_GENERATED_VIA_STATS_TASK` to `true`, `COLUMN_STATS_ACCURATE` will be `true`. For example, 
```SQL
alter table t1 set TBLPROPERTIES ('numRows'='2', 'STATS_GENERATED_VIA_STATS_TASK'='true');
```
The results will be like
```
Table Parameters:	 	 
	COLUMN_STATS_ACCURATE	true                
	last_modified_by    	root                
	last_modified_time  	1471635407          
	numFiles            	1                   
	numRows             	2                   
	rawDataSize         	-1                  
	totalSize           	2                   
	transient_lastDdlTime	1471635407    
```

```